### ISSUE TITLE
,py file - in memory df viewer not available

### ISSUE THREAD

**Issue by droy-proxsys** [microsoft/vscode-data-wrangler#253]
## Environment data

-   VS Code version: 1.92
-   Data Wrangler Extension version (available under the Extensions sidebar): 1.6
-   Jupyter Extension version (available under the Extensions sidebar): 2024.7.0
-   Python Extension version (available under the Extensions sidebar): 3.12
-   OS (Windows | Mac | Linux distro) and version: windows 11

## Expected behaviour

view an in mem dataframe

## Actual behaviour

![image](https://github.com/user-attachments/assets/a1bcae54-4082-4b3f-8481-5db4487ea76b)

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

I do not do anything with jupyter notebooks, I only use .py files.
1. while debugging have a break points set right after making a dataframe
2. right click and and say view in data viewer
![image](https://github.com/user-attachments/assets/9bc7f663-148c-43d3-b0f6-ac0f04a039f9)

## Logs

no error just does not work. Jupyter says you are supposed to be the drop in replacement, but I am not believing them right now...

**Comment by pwang347** [comment]
Hi @droy-proxsys! We are sorry that you're running into this issue. Based on your description, it seems like this might be related to https://github.com/microsoft/vscode-jupyter/issues/15877 - do the steps described in the thread resolve the issue for you?

Separately, if you can share the following information it would help us investigate this further:
1. Does this happen for any dataset or only specific ones? E.g. does the following simple DataFrame load for you while debugging?
```python
df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
```

2. When the viewer doesn't launch, do you see any errors in the console log? You can access that through the following menu:
<img width="372" alt="image" src="https://github.com/user-attachments/assets/58cbd070-575a-4a4e-9967-2f79f3b854e5">

And then in the "Console" tab:
<img width="313" alt="image" src="https://github.com/user-attachments/assets/e54d079b-b6e4-4f29-8a5b-c10366129861">

3. Could you please set the Jupyter logging level to "debug" and then reproduce the issue again? You can find this setting by searching for "jupyter logging" from the settings UI.
<img width="561" alt="image" src="https://github.com/user-attachments/assets/0701e7e5-bf48-4e4b-a345-05e3daa696f0">

We would then appreciate if you could then copy and attach the logs in the Output panel here:
<img width="486" alt="image" src="https://github.com/user-attachments/assets/bd17ff2f-9c76-4a69-977a-ba92f95fc680">

Thank you!

**Comment by droy-proxsys** [comment]
This is not related to the mentioned issue, if anything it is most related to https://github.com/microsoft/vscode-jupyter/issues/15486 as they have depreciated their support for of dataviewer. So now when you try to view it you are met with the following.
![image](https://github.com/user-attachments/assets/734fd1a7-30ea-4d29-b577-b689ce8a12b3)

Of which, data wrangler is what comes up.
![image](https://github.com/user-attachments/assets/d697526a-74c6-43f1-99c7-7310ad83fec2)

"When the viewer doesn't launch, do you see any errors in the console log? You can access that through the following menu:"
-> It just does not even get that far. nothing fires at all.

Does this happen for any dataset or only specific ones? E.g. does the following simple DataFrame load for you while debugging?
--> This is only during debugging of a .py file. Does not matter what data is in it at all.
![image](https://github.com/user-attachments/assets/89d438b1-c74a-4b7d-baf9-c5fc2e16723b)
![image](https://github.com/user-attachments/assets/03543617-bd86-4b5c-85ed-d4feb7c10bcf)

debug log - though it does not do anything... 
``` 
Visual Studio Code (1.92.0, undefined, desktop)
Jupyter Extension Version: 2024.7.0.
Python Extension Version: 2024.12.2.
Pylance Extension Version: 2024.8.1.
Platform: win32 (x64).
Workspace folder c:\GitHub\__sandbox__, Home = c:\xxxxxxxxxxxxxxxxxxxxxxxx
08:29:31.851 [info] Ending debug session 6eeba859-b094-4de8-b055-9958bcbfe199
08:29:31.851 [info] Ending debug session 6eeba859-b094-4de8-b055-9958bcbfe199
08:31:24.886 [debug] Evaluating in debugger : 96033708-996b-40cf-91b2-8f54ef5bdbab: _VSCODE_getVariable("info", True, df)
```


It very well could be that this add in does not have that feature at all. If that is the case without the data viewer being available any more I almost say you should update your description to annotate that it is not available as I would expect other data devs to come knocking as well.

**Comment by pwang347** [comment]
Hi @droy-proxsys, thank you for the reply and for providing more information.

Data Wrangler does support the ability to view DataFrames in memory while debugging Python files, ie. the scenario you are showing is supposed to work. Note that the "View Value in Data Viewer" button is from the Jupyter extension, so I think it's possible that the routing there is not working and our extension never gets called (the issue I referred to above is an example of it). We will share a new pre-release version tomorrow that will provide more notifications about the scenario so that we can check to be sure that there aren't any fixes that we can make on our end.

In the meantime, could you please try the following workarounds to see if they can unblock your workflow:
* Debug using the interactive window instead (e.g. running the command `Jupyter: Debug Current File in Interactive Window` after opening the command palette with `ctrl+shift+P`), or type `# %%` above your Python code and click `Debug Cell` above
<img width="261" alt="image" src="https://github.com/user-attachments/assets/ffaaaf44-5c17-4288-9dcd-35b7d93e6ed2">


* Try using a different Python interpreter by clicking on the bottom right status bar selector
<img width="236" alt="image" src="https://github.com/user-attachments/assets/a0c1389c-35e2-4c13-85e3-06db20b1f24b">

Please let me know if these help. Thanks!

**Comment by droy-proxsys** [comment]
Yeah same thing sadly...
![image](https://github.com/user-attachments/assets/9779e59d-7df4-47fa-9af4-7c7836713513)

I will say there is another issue that this showed as well...
![image](https://github.com/user-attachments/assets/ae8120eb-f50d-49ba-9117-c5155153ee5c)

When I click on the 'View data' button it shows the following alert despite having everything installed already.
![image](https://github.com/user-attachments/assets/13e09795-27a4-4ef9-b29d-202dabcdfb6d)

When i click yes, this is where it sits doing nothing... (I let it sit for 10+ min while I made fresh nectar of life, coffee)
![image](https://github.com/user-attachments/assets/c8f0d6d2-367b-4662-8d1f-9e3414a6f749)

no matter what when i click on that view data button it is a death loop of 'go install' nothing installs...


Jupyter logs:

06:28:04.112 [debug] Initializing __file__ in setFileInKernel with file:///c%3A/GitHub/__sandbox__/tst.py for mode multiple
06:28:04.112 [trace] Executing silently Code (idle) = __file__ = 'c:\\GitHub\\__sandbox__\\tst.py'
06:28:04.117 [trace] Executing silently Code (idle) = def _VSCODE_getVariable(what_to_get, is_debugging, *args):\n# Query Jupyter server for the info
06:28:04.119 [trace] Executing silently Code (completed) = __file__ = 'c:\\GitHub\\__sandbox__\\tst.py' with 0 output(s)
06:28:04.152 [trace] Executing silently Code (completed) = def _VSCODE_getVariable(what_to_get, is_debugging, *args):\n# Query Jupyter server for the info with 1 output(s)
06:28:19.453 [info] Generated code for 1 = <ipython-input-1-7ce80b8b19b0> with 7 lines
06:28:19.487 [info] start debugging
06:28:19.487 [info] enable debugger attach
06:28:19.488 [trace] Executing silently Code (idle) = import sys as _VSCODE_sys\n_VSCODE_sys.path.extend(['c:\\Users\\~\\.vscode\\extensions\\ms-p
06:28:19.500 [trace] Executing silently Code (completed) = import sys as _VSCODE_sys\n_VSCODE_sys.path.extend(['c:\\Users\\~\\.vscode\\extensions\\ms-p with 0 output(s)
06:28:19.500 [info] Appending paths: undefined
06:28:19.500 [trace] Executing silently Code (idle) = import debugpy;debugpy.listen(('localhost', 0))
06:28:20.402 [trace] Executing silently Code (completed) = import debugpy;debugpy.listen(('localhost', 0)) with 1 output(s)
06:28:20.402 [info] connected to notebook during debugging
06:28:20.567 [trace] Executing silently Code (idle) = import debugpy;debugpy.wait_for_client()
06:28:20.594 [trace] Executing silently Code (completed) = import debugpy;debugpy.wait_for_client() with 0 output(s)
06:28:20.595 [info] import startup: undefined
06:28:20.595 [trace] Executing silently Code (busy) = from debugpy import trace_this_thread;trace_this_thread(False)
06:28:20.621 [trace] Executing silently Code (completed) = from debugpy import trace_this_thread;trace_this_thread(False) with 0 output(s)
06:28:20.642 [trace] Executing silently Code (idle) = from debugpy import trace_this_thread;trace_this_thread(True)
06:28:20.645 [trace] Cell Index:3 sent to kernel
06:28:20.662 [trace] Executing silently Code (completed) = from debugpy import trace_this_thread;trace_this_thread(True) with 0 output(s)
06:28:20.678 [trace] Start cell 3 execution @ 1723116500678 (clear output)
06:28:20.679 [debug] Kernel acknowledged execution of cell 3 @ 1723116500678
06:28:32.183 [debug] Evaluating in debugger : b4eb6d0e-6a91-455a-a81e-a84220590b4b: _VSCODE_getVariable("info", True, file)
06:28:34.088 [debug] Evaluating in debugger : b4eb6d0e-6a91-455a-a81e-a84220590b4b: _VSCODE_getVariable("info", True, df)
06:28:34.125 [debug] Evaluating in debugger : b4eb6d0e-6a91-455a-a81e-a84220590b4b: _VSCODE_getVariable("info", True, file)
06:33:27.809 [trace] Cell 3 completed in 307.13s (start: 1723116500678, end: 1723116807808)
06:33:27.811 [trace] Cell 3 executed successfully
06:33:27.813 [info] stop debugging
06:33:27.813 [trace] Executing silently Code (idle) = from debugpy import trace_this_thread;trace_this_thread(False)
06:33:27.829 [trace] Executing silently Code (completed) = from debugpy import trace_this_thread;trace_this_thread(False) with 0 output(s)
06:33:27.890 [trace] Executing silently Code (idle) = def _VSCODE_getVariable(what_to_get, is_debugging, *args):\n# Query Jupyter server for the info
06:33:27.899 [trace] Executing silently Code (idle) = def _VSCODE_getVariable(what_to_get, is_debugging, *args):\n# Query Jupyter server for the info
06:33:27.929 [trace] Executing silently Code (completed) = def _VSCODE_getVariable(what_to_get, is_debugging, *args):\n# Query Jupyter server for the info with 1 output(s)
06:33:27.939 [info] Ending debug session b4eb6d0e-6a91-455a-a81e-a84220590b4b
06:33:27.939 [info] Ending debug session b4eb6d0e-6a91-455a-a81e-a84220590b4b
06:33:27.965 [trace] Executing silently Code (completed) = def _VSCODE_getVariable(what_to_get, is_debugging, *args):\n# Query Jupyter server for the info with 1 output(s)

**Comment by pwang347** [comment]
@droy-proxsys Interesting, this looks a lot like another issue I've seen before. Could you please check if you have this setting and if so to follow the steps here? https://github.com/microsoft/vscode-data-wrangler/issues/129#issuecomment-1922177014

Thanks!

**Comment by droy-proxsys** [comment]
Well played @pwang347! That was indeed the trick. I'm going to throw it below in case someone else comes hunting. Thanks for your help! (I have walked back my rage rating on vs market too... )



Future seekers, (or TLDR's) if the following is in you setting.json file - remove it OR add "ms-toolsai.datawrangler": 1

```
    "extensions.experimental.affinity": {
        "ms-toolsai.jupyter": 1,
        "ms-toolsai.jupyter-renderers": 1,
        "ms-python.python": 1,
        "ms-python.vscode-pylance": 1,
    },
```

**Comment by UdaykiranEstari** [comment]
@droy-proxsys Have you found any workaround to view in-memory dataframes while debugging .py files
